pkg/scheduler/plugins/binpack/binpack.go

```diff
@@ -0,0 +1,230 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package binpack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
+
+	"volcano.sh/volcano/pkg/scheduler/api"
+	"volcano.sh/volcano/pkg/scheduler/framework"
+)
+
+const (
+	// PluginName indicates name of volcano scheduler plugin.
+	PluginName = "binpack"
+)
+
+const (
+	// BinpackWeight is the key for providing Binpack Priority Weight in YAML
+	BinpackWeight = "binpack.weight"
+	// BinpackCPU is the key for weight of cpu
+	BinpackCPU = "binpack.cpu"
+	// BinpackMemory is the key for memory of cpu
+	BinpackMemory = "binpack.memory"
+
+	// BinpackResources is the key for additional resource key name
+	BinpackResources = "binpack.resources"
+	// BinpackResourcesPrefix is the key prefix for additional resource key name
+	BinpackResourcesPrefix = BinpackResources + "."
+
+	resourceFmt = "%s[%d]"
+)
+
+type priorityWeight struct {
+	BinPackingWeight    int
+	BinPackingCPU       int
+	BinPackingMemory    int
+	BinPackingResources map[v1.ResourceName]int
+}
+
+func (w *priorityWeight) String() string {
+	length := 3
+	if extendLength := len(w.BinPackingResources); extendLength == 0 {
+		length++
+	} else {
+		length += extendLength
+	}
+	msg := make([]string, 0, length)
+	msg = append(msg,
+		fmt.Sprintf(resourceFmt, BinpackWeight, w.BinPackingWeight),
+		fmt.Sprintf(resourceFmt, BinpackCPU, w.BinPackingCPU),
+		fmt.Sprintf(resourceFmt, BinpackMemory, w.BinPackingMemory),
+	)
+
+	if len(w.BinPackingResources) == 0 {
+		msg = append(msg, "no extend resources.")
+	} else {
+		for name, weight := range w.BinPackingResources {
+			msg = append(msg, fmt.Sprintf(resourceFmt, name, weight))
+		}
+	}
+	return strings.Join(msg, ", ")
+}
+
+type binpackPlugin struct {
+	// Arguments given for the plugin
+	weight priorityWeight
+}
+
+//New function returns prioritizePlugin object
+func New(aruguments framework.Arguments) framework.Plugin {
+	weight := calculateWeight(aruguments)
+	return &binpackPlugin{weight: weight}
+}
+
+func calculateWeight(args framework.Arguments) priorityWeight {
+	/*
+	   User Should give priorityWeight in this format(binpack.weight, binpack.cpu, binpack.memory).
+	   Support change the weight about cpu, memory and additional resource by arguments.
+
+	   actions: "enqueue, reclaim, allocate, backfill, preempt"
+	   tiers:
+	   - plugins:
+	     - name: binpack
+	       arguments:
+	         binpack.weight: 10
+	         binpack.cpu: 5
+	         binpack.memory: 1
+	         binpack.resources: nvidia.com/gpu, example.com/foo
+	         binpack.resources.nvidia.com/gpu: 2
```

Comment:
s/`binpack.resources.nvidia.com/gpu`/`binpack.nvidia.com/gpu`/? Then it is same as cpu/memory, it is just a suggestion.

Reply:
In my opinion, all additional resource is listed in `binpack.resources`, so the key for weight of resource should have prefix of it.